### PR TITLE
[RAPTOR-5540] support all datatypes

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
This is to match the validation used in DR for the schema.  NO runtime validation in DRUM is supported for some types, such as audio, at this time.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 1]
```diff
@@ -194,14 +210,21 @@ class DataTypes(BaseValidator):
     """Validation related to data types.  This is common between input and output."""
 
     def __init__(self, condition, values):
+        # We currently do not support DRUM validation for these values, but they are supported in DataRobot
+        self._SKIP_VALIDATION = {Values.AUDIO, Values.COUNT_DICT, Values.GEO, Values.TARGET_ONLY}
+        values = list(set(values) - self._SKIP_VALIDATION)
+        if len(values) == 0:
+            logger.info(
+                "Values specified do not have runtime validation in DRUM, only within DataRobot."
```
Comment (4dahalibut) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:218]: can you add the specific value in this info message.

**Review by 4dahalibut — APPROVED** [review]
LGTM but please make single change.

**Comment by 4dahalibut** [comment]
Have you manually tested using these data types w datarobot yet?

**Comment by scottp-ml** [comment]
I have not built a task for each one.

File: VALIDATION-SCHEMA.md [hunk 2]
```diff
@@ -20,8 +20,8 @@ list for input requirements and for output requirements unless noted otherwise b
 The `data_types` field is used to specify those datatypes that are specifically expected, or those that
 are specifically disallowed.  Only a single entry is supported for this field in each specification.  
 The conditions used for data_types are:
-- EQUALS: A single data type is expected and must match the provided value for all columns.
-- IN: All of the listed data types are expected in the dataframe.  An error will occur if one is missing or unexpected types are found.
+- EQUALS:All of the listed data types are expected in the dataframe.  An error will occur if one is missing or unexpected types are found.
+- IN: The types are supported, but not all are required to be present.
```
Comment (eric-s-s) [inline VALIDATION-SCHEMA.md:24]: All of the listed data types are supported  (mirroring the equals language)

**Review by eric-s-s — CHANGES_REQUESTED** [review]
somewhere around: `tests.drum.unit.test_typeschema_validation.TestSchemaValidator.test_data_types` , please write test for the following cases:

- Values.DATE_DURATION
- all operators with each of the types we don't validate
- all operators that take multiple types with 
  - 2 types we don't validate
  - a type we do validate and a type we don't

**Review by eric-s-s — APPROVED** [review]
:+1: looks good!

**Pull request merged**